api/src/main/java/com/epam/pipeline/manager/cluster/performancemonitoring/ResourceMonitoringManager.java

```diff
@@ -118,11 +122,12 @@ public void removeOldIndices() {
     public void monitorResourceUsage() {
         List<PipelineRun> runs = pipelineRunManager.loadRunningPipelineRuns();
 
-        processIdleRuns(runs);
-        processOverloadedRuns(runs);
+        final List<PipelineRun> runsToUpdateTags = processIdleRuns(runs);
```

Comment:
lets updates tags for overloaded and idled runs separately, to increase code readability